### ISSUE TITLE
New Journal overwrites draft tiddler

### ISSUE THREAD

**Issue by Arlen22** [Jermolene/TiddlyWiki5#2931]
Clicking New Journal overwrites the tiddler currently being edited if it has the same title. Even if the draft title has already been changed but it has not been saved yet with the new title. The title field of the draft does not change, but the text field gets set to the default, or cleared if there is no default.

**Comment by tobibeer** [comment]
@Arlen22: Possible this is fixed?

**Comment by Dialga** [comment]
I am tempted to open a new issue because I've just lost a long journal entry that I wrote earlier in the morning due to encountering this problem.
1. create a new journal for today with some text, save and exit editor
2. create a new journal and change the date to last week, notice today's journal entry is loaded, edit, save and close editor.
Now the original tiddler for today is overwritten and lost, I cannot fathom why using the new journal button to create an old journal or future entry with a completely different date will delete the existing journal entry for today. This is incredibly frustrating.

**Comment by Arlen22** [comment]
>I cannot fathom why using the new journal button to create an old journal or future entry with a completely different date will delete the existing journal entry for today. This is incredibly frustrating.

Because a new draft is being created with the same name and opened for editing. It does not check for existing drafts like the New Tiddler button **_presumably_** does. Might want to make sure the New Tiddler button works properly while we're at it.

**Comment by pmario** [comment]
>I am tempted to open a new issue because I've just lost a long journal entry that I wrote earlier in the morning due to encountering this problem.

I'm sorry to read, that you lost some data. 

You should have created a new issue. They are related, but not exactly the same and the OP actually seems to be fixed. 

So imo the issue should have been closed already.

**Comment by pmario** [comment]
> 2\. ... I cannot fathom why using the new journal button to create an old journal or future entry with a completely different date will delete the existing journal entry for today. This is incredibly frustrating.

The problem here is, that if you open the journal tiddler with the "New Journal" button, it actually is an "Open Journal" button, that depends on the "title template" that can be adjusted in the ControlPanel. 

The default format is: `DDth MMM YYYY`, which will re-open the same journal tiddler for one day. 

If a tiddler is switched to "edit mode" a new tiddler named. `Draft of 'xxxx'` is "cloned" from the existing `xxxx` tiddler. The draft tiddler is "connected" to the tiddler from which it originates. 

If you click the "cancel icon" you expect the "draft tiddler" to be thrown away and no changes should be made to the original tiddler. right?

If you click the "bin icon", you expect the tiddler _and_ the draft tiddler to be deleted. right?

If you rename a tiddler, you expect the original tiddler to be renamed. right?

The expectation, that opening a tiddler and renaming it, will create 2 tiddlers is wrong. ... right?

So if you need a new journal for a tiddler 1 week ago, you need to create a "standard" new tiddler and adjust everything accordingly. 

I'm sorry, that you lost data. .... But it has been some wrong expectations that has lead to that problem. Not necessarily the system itself.

**Comment by pmario** [comment]
> The default format is: `DDth MMM YYYY`, which will re-open the same journal tiddler for one day.

I personally use this format for my journal tiddlers: `YYYY.0MM.0DD 0hh:0mm:0ss` ... Which is a completely different workflow. ... 

For me journal tiddlers are temporary tiddlers, that will be converted into something else ... most of the time.

I manly use them, to avoid tiddler names like "New Tiddler 1", "New Tiddler 2" ... and so on. Usually for me it's relatively safe to delete old "New Tiddler X" without further checks ... but I won't delete journal tiddlers, without a decent check if they can be deleted. 

Just some thoughts

**Comment by Dialga** [comment]
> The problem here is, that if you open the journal tiddler with the "New Journal" button, it actually is an "Open Journal" button, that depends on the "title template" that can be adjusted in the ControlPanel.

That is not what I was expecting here, I want to create a new journal entry by clicking the "New Journal" button so it should create a new journal entry, clicking "New Journal" should not open the today's journal entry, and if it already exists append an incremental number to it as would be a new entry for today.
I was expecting the "New Journal" button to behave as if using a paper diary, if I want to retrospectively create a journal entry I would go to that page and write, not open today's entry, erase it and then backdate it.

> The expectation, that opening a tiddler, changing the text and renaming it, will create 2 tiddlers is wrong. ... right?

Yes, because in this case it is _opening an existing_ tiddler.

This is frustrating because I have used the "New Journal" button before in the past to create multiple past entries on the same day, and just found out this **only** works if there wasn't already a journal created for today.

IMO the "New Journal" button should do exactly that: create a new journal tiddler, not "create or update existing journal tiddler". Because in its current form it is bit misleading, perhaps "New Journal" should use another adjective or the mousetip should say "Create or open journal tiddler". But otherwise, lesson learnt.

**Comment by pmario** [comment]
>IMO the "New Journal" button should do exactly that: create a new journal tiddler, not "create or update existing journal tiddler". 

Yea. That's one possible opinion -- and [I share it](https://github.com/Jermolene/TiddlyWiki5/pull/5614). But that's _not_ the default behaviour. 

The default behaviour is also valid: 

 - 1 Journal tiddler for 1 day. 
 - Content is appended to that tiddler. 

-----

If we want to change that, we need to use a different title-template eg: `YYYY.0MM.0DD 0hh:0mm:0ss`. That's the one I use. This format "naturally" sorts by title if shown in lists. I can create as many journals as I like and have no renaming problems with it.

**Comment by Jermolene** [comment]
> I am tempted to open a new issue because I've just lost a long journal entry that I wrote earlier in the morning due to encountering this problem.
> 
> 1. create a new journal for today with some text, save and exit editor
> 2. create a new journal and change the date to last week, notice today's journal entry is loaded, edit, save and close editor.
>    Now the original tiddler for today is overwritten and lost, I cannot fathom why using the new journal button to create an old journal or future entry with a completely different date will delete the existing journal entry for today. This is incredibly frustrating.

Hi @Dialga apologies for coming late to this. I'm sorry you lost data. Annoyingly, I am not sure that there is a clear solution to this. The actions in (2) describe how to rename a tiddler in TiddlyWiki: bring it up for editing, and change the input box for the title.

It is reasonable to expect that those actions would instead create a new journal for the different date. But that would mean that renaming worked differently in this particular case. It's hard to see how the system can distinguish this special case from the normal case of renaming a tiddler. And I guess there are legitimate reasons for wanting to rename a journal tiddler, too.

> IMO the "New Journal" button should do exactly that: create a new journal tiddler, not "create or update existing journal tiddler". Because in its current form it is bit misleading, perhaps "New Journal" should use another adjective or the mousetip should say "Create or open journal tiddler". But otherwise, lesson learnt.

I think there's enough variation in how different users approach journaling that it might be reasonable to add a checkbox to control this behaviour. I'd just worry that it might not be very discoverable.

**Comment by CodaCodr** [comment]
Completely agree with @Dialga's reasoning - paraphrased, "new means new". The <big><kbd>+</kbd></big> also has accumulated baggage that drives my thinking -- I'm _adding_ / _creating_, not editing.

So, working with today a second time on tiddlywiki.com,

```
title: 6th May 2022 (1)
```

Third:

```
title: 6th May 2022 (2)
```
etc.

If I want to edit a journal tiddler, it's a tiddler -- find it and edit it.

**Comment by Jermolene** [comment]
> Completely agree with @Dialga's reasoning - paraphrased, "new means new". The + also has accumulated baggage that drives my thinking -- I'm _adding_ / _creating_, not editing.

Right, but the current behaviour is intentional. So the solution cannot be to unilaterally change how the existing button works, we need to find a backwards compatible way to accommodate these conflicting needs.

**Comment by CodaCodr** [comment]
New, additional button with different icon? Well doc'd to make it clear, _this is new behaviour_?

**Comment by pmario** [comment]
>... this is new behaviour?

No. It has been like that since the beginning. ... A new button would be "a new behaviour".

**Comment by CodaCodr** [comment]
@pmario I think you missed the point - that _was_ my point. Let me flesh it out for you:

Proposal: Add a new button with different icon. Use control panel to configure _which_ button is in effect, the old one or the new one. The old one uses the old/current behavior. The new button works like "New Tiddler" but with journaling applied a la https://github.com/Jermolene/TiddlyWiki5/issues/2931#issuecomment-1119833134

**Comment by CodaCodr** [comment]
https://talk.tiddlywiki.org/t/create-new-journal-tiddler-broken/3252

**Comment by Jermolene** [comment]
Hi @CodaCodr having two separate buttons presents quite a cognitive load for users:

* Understanding what to do if they don't see the button they expect
* Being able to make an informed decision in advance as to which button to use
* Distinguishing between the two buttons and communicating their function

That makes it not a good solution for the core. But it's exactly the right solution for plugin authors: to add a completely independent button that cannot interfere with subsequent core updates.

For the core, perhaps it makes more sense to keep a single button, but make its action configurable with a dialogue that explains the options. Perhaps the configuration dialogue could be surfaced more prominently somewhere else as well as in control panel. For example, perhaps we could add a segment to the edit template for journal tiddlers that contains the configuration dialogue.

**Comment by pmario** [comment]
>... For example, perhaps we could add a segment to the edit template for journal tiddlers that contains the configuration dialogue.

With the new cascade mechanism, this should be relatively straight forward. The code could be similar to the blue "warnig box" for tiddler titles. 

![image](https://user-images.githubusercontent.com/374655/167407428-28eaf070-a719-4619-a44d-7366decea2d3.png)

**Comment by gabrielmontagne** [comment]
There's an inconsistency/bug in the current implementation (runnging 5.3.6), please see: 

https://talk.tiddlywiki.org/t/how-can-i-prevent-journal-data-loss/14075/9

That is, 

If you click the Journal for today and edit it, but not yet commit it (keeping it as a draft), if you click again the journal button: 

* if you haven't configured a "New Journal Text", meaning: there's nothing on [New Journal Text](https://tiddlywiki.com/#%24%3A%2Fconfig%2FNewJournal%2FText), you'll be taken to your started Journal entry draft, and you'll be able to continue working on it from where you started.
* If you do have a "New Journal Text", whatever you had on your draft will be wiped out and replaced with the template.  The text will be unrecoverable.

I understand the semantic of "New Journal" button, but some of us might think of it as "Go to/Work on my Journal for today" (even Journal is etymologically a kind of diary), which works fine (as I'd expect) if you've saved your draft at least once (or if you don't have a template defined). 

I know one can define the journal name to include time, so in that case you want it to create new entries for new times (though, one could argue that if you go to only by hour of the day (my 4pm entry, my 5pm entry) one would expect to keep working on that same entry until the clock jumped to the next hour.

Again, it's already inconsistent: if a Journal remains the same on subsequent accesses after being saved once, then the draft should also remain consistent while being accessed repeatedly.
